airflow/api_fastapi/views/public/dags.py

```diff
@@ -92,6 +93,28 @@ async def get_dags(
     )
 
 
+@dags_router.get(
+    "/tags",
+    response_model=list[DAGTagResponse],
+    responses=create_openapi_http_exception_doc([400, 401, 403]),
+)
+async def get_dag_tags(
+    tags: QueryTagsFilter,
+    session: Annotated[Session, Depends(get_session)],
+) -> list[DAGTagResponse]:
+    """Get all DAG tags."""
+    dag_tag_names = session.query(distinct(DagTag.name)).order_by(DagTag.name).all()
+    if not dag_tag_names:
+        return []
+    selected_dag_tags = {}
+    if tags.value:
+        selected_dag_tags = {tag: True for tag in tags.value}
+    return [
+        DAGTagResponse(name=tag_name_row[0], selected=selected_dag_tags.get(tag_name_row[0], False))
+        for tag_name_row in dag_tag_names
```

Comment:
I do not understand the `selected` field in the response. We query tags with a filter on the names, then we should get back in response a collection of tags, maybe paginated if the request says so. 

This should be more homogenous to other list endpoints that the API have, you can take a look at them to implement something similar. (get_dags for instance)

Reply:
According to the issue https://github.com/apache/airflow/issues/42713, the response schema should match the format defined here: https://github.com/apache/airflow/blob/main/airflow/www/views.py#L1035. The `selected` field indicates whether the tag is selected based on the query parameters in all `dag_tags`.

Reply:
However, I believe this can be handled on the frontend (by storing the user’s selected state there, instead of passing tags through the API to get the `selected` fields). It’s not necessary to migrate the `selected` logic from the legacy API.

Reply:
I think this is an area where we need to adjust the implementation a bit. Before with FAB a lot of frontend logic was done on the webserver. Now, we should just send a list of tags to the UI and it handles all its own selected logic.

So let's drop `selected`. But in addition to pagination, I think we'll need a `tag_name_pattern` filter param.